src/app/static/src/app/utils.ts

```diff
@@ -247,48 +252,27 @@ export const validateEmail = (test: string): boolean => {
 export const versionLabel = (version: Version) => `v${version.versionNumber}`;
 
 export const updateForm = (oldForm: DynamicFormMeta, newForm: DynamicFormMeta): DynamicFormMeta => {
-    const oldSectionLabels = oldForm.controlSections.map(c => c.label);
-
-    newForm.controlSections = newForm.controlSections.map(s => {
-        const oldIndex = oldSectionLabels.indexOf(s.label);
-        if (oldIndex == -1) {
-            return s
-        } else {
-            return updateSection(oldForm.controlSections[oldIndex], s)
-        }
-    });
-
-    return newForm
-};
-
-function updateSection(oldSection: DynamicControlSection, newSection: DynamicControlSection) {
-
-    const oldGroupLabels = oldSection.controlGroups.map(g => g.label);
-    newSection.controlGroups = newSection.controlGroups.map(g => {
-        const oldGroupIndex = oldGroupLabels.indexOf(g.label);
-        if (oldGroupIndex == -1) {
-            return g
-        } else {
-            return updateGroup(oldSection.controlGroups[oldGroupIndex], g)
-        }
+    const selectedOptions = {} as Dict<string | string[] | number | null | undefined>;
```

Comment:
This is now a thing of beauty... but slightly surprised we (still) don't have any tests for this? Unless I'm looking in the wrong place.

Reply:
It is (and was) tested via the modelOptions mutations tests - but you're right, it's worth adding some explicit tests for this method.

Reply:
Ah ok - sorry